charts/csi-cloudscale/templates/daemonset.yaml

```
@@ -0,0 +1,107 @@
+        role: csi-cloudscale
+    spec:
+      priorityClassName: system-node-critical
+      serviceAccount: {{ include "csi-cloudscale.node-service-account-name" . }}
+      hostNetwork: true
+      containers:
+        - name: csi-node-driver-registrar
+          image: {{ .Values.driverRegistrar.image.repository }}:{{ .Values.driverRegistrar.image.tag }}
+          imagePullPolicy: {{ .Values.driverRegistrar.image.pullPolicy }}
+          args:
+            - "--v=5"
+            - "--csi-address=$(ADDRESS)"
+            - "--kubelet-registration-path=$(DRIVER_REG_SOCK_PATH)"
+          lifecycle:
+            preStop:
+              exec:
+                command: ["/bin/sh", "-c", "rm -rf /registration/csi.cloudscale.ch /registration/csi.cloudscale.ch-reg.sock"]
+          env:
+            - name: ADDRESS
+              value: /csi/csi.sock
+            - name: DRIVER_REG_SOCK_PATH
+              value: /var/lib/kubelet/plugins/csi.cloudscale.ch/csi.sock
+            - name: KUBE_NODE_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: spec.nodeName
+          {{- with .Values.driverRegistrar.resources }}
+          resources:
+{{ toYaml . | indent 12 }}
+          {{- end }}
+          volumeMounts:
+            - name: plugin-dir
+              mountPath: /csi/
+            - name: registration-dir
+              mountPath: /registration/
+        - name: csi-cloudscale-plugin
+          image: {{ .Values.node.image.repository }}:{{ .Values.node.image.tag }}
+          imagePullPolicy: {{ .Values.node.image.pullPolicy }}
+          args :
+            - "--endpoint=$(CSI_ENDPOINT)"
+            - "--url=$(CLOUDSCALE_API_URL)"
+          {{- with .Values.node.resources }}
+          resources:
+{{ toYaml . | indent 12 }}
+          {{- end }}
+          env:
+            - name: CSI_ENDPOINT
+              value: unix:///csi/csi.sock
+            - name: CLOUDSCALE_API_URL
+              value: {{ .Values.cloudscale.apiUrl }}
+            - name: CLOUDSCALE_MAX_CSI_VOLUMES_PER_NODE
+              value: {{ .Values.node.max_csi_volumes_per_node }}
+            - name: CLOUDSCALE_ACCESS_TOKEN
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "csi-cloudscale.api-token-name" . }}
+                  key: access-token
+          securityContext:
+            privileged: true
+            capabilities:
+              add: ["SYS_ADMIN"]
+            allowPrivilegeEscalation: true
+          volumeMounts:
+            - name: plugin-dir
+              mountPath: /csi
+            - name: pods-mount-dir
+              mountPath: /var/lib/kubelet
+              # needed so that any mounts setup inside this container are
+              # propagated back to the host machine.
+              mountPropagation: "Bidirectional"
+            - name: device-dir
+              mountPath: /dev
+            - name: tmpfs
+              mountPath: /tmp
+      volumes:
```

We currently have some logic to add [`tolerations`](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/) for the node driver DaemonSet, to allow it to run on nodes which are tainted to prevent regular workloads to run on them. It would be nice if the Helm chart would provide support for that, something like

```suggestion
      {{- with .Values.node.tolerations }}
      tolerations:
{{ toYaml . | indent 8 }}
      {{- end }}
      volumes:
```

---

Added `node.tolerations` value.